### PR TITLE
docs(router): change description of parameter usage for navigation functions

### PR DESCRIPTION
**Pull request by vincecampanale**
…navigateByUrl to reduce confusion and highlight contrast

@kapunahelewong 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #20421
https://github.com/angular/angular/issues/20421

## What is the new behavior?
Explicitly describe behavior of each method and intention of parameters.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/router/src/router.ts [hunk 1]
```diff
@@ -441,8 +441,10 @@ export class Router {
    * router.navigateByUrl("/team/33/user/11", { skipLocationChange: true });
    * ```
    *
-   * In opposite to `navigate`, `navigateByUrl` takes a whole URL
-   * and does not apply any delta to the current one.
+   * Since `navigateByUrl` takes an absolute url as the first parameter, 
+   * it will not apply any delta to the current url and ignores any properties 
+   * in the second parameter (the `NavigationExtras`) that would change the
+   * provided url.
```
Comment (kapunahelewong) [inline packages/router/src/router.ts]: Would you change:

url --> URL

File: packages/router/src/router.ts [hunk 2]
```diff
@@ -470,8 +472,8 @@ export class Router {
    * router.navigate(['team', 33, 'user', 11], {relativeTo: route, skipLocationChange: true});
    * ```
    *
-   * In opposite to `navigateByUrl`, `navigate` always takes a delta that is applied to the current
-   * URL.
+   * The first parameter of `navigate` is a delta to be applied to the current url 
+   * or the one provided in the `relativeTo` property of the second parameter (the `NavigationExtras`).
    */
```
Comment (kapunahelewong) [inline packages/router/src/router.ts]: And here, would you change these two things:

`navigate` --> `navigate()`
url --> URL

**Review by kapunahelewong — CHANGES_REQUESTED** [review]
This is very helpful, thank you @vincecampanale. Just tiny requests.

My apologies for missing the notification email.

**Comment by vincecampanale** [comment]
@kapunahelewong  No problem! Thanks for getting back to me. I made the changes you requested along with a couple of others based on yours.

**Review by kapunahelewong — APPROVED** [review]
Thank you @vincecampanale! LGTM from the copy standpoint.

**Review by mhevery — APPROVED** [review]

**Comment by IgorMinar** [comment]
I've rebased this.

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**